File: C#_Projects/Magaza/YapıMalzemeleri.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Magaza
{
    public class YapıMalzemeleri : Anasayfa
    {
        string[] urunler = { "çekiç", "balta", "çivi", "masa", "sunta", "boya", "testere", "metre" };
        string kampanyaliUrun;

        public YapıMalzemeleri(int kodu, string adi, float agirligi, string mensei) : base(kodu, adi, agirligi, mensei)
        {
        }

        public override void genelOzellikleriGoster()
        {
            Console.WriteLine(String.Format("Bu yapı malzemesinin ürün kodu: {0}, ürün adı {1}, ürün ağırlığı: {2} kilogram, urunun menşei ise {3} gözükmektedir", urunAdi, urunKodu, urunAgirligi, urunMensei));
        }

        public override void nasilSatilacak()
        {
            for (int i = 0; i < urunler.Length; i++)
            {

                if (urunler[i] == "çekiç" || urunler[i] == "çivi" || urunler[i] == "sunta")
                {
                    kampanyaliUrun = urunler[i];
                    Console.WriteLine("Yapı malzemesi bölümünün " + kampanyaliUrun + " ürününde 20.01.2023 tarhine kadar kampanya var");
                }
            }
        }
    }
}

File: Javascript_Projects/Magaza/Yapı_Malzemeleri.js
import { Anasayfa } from "./Anasayfa.js";

export class Yapı_Malzemeleri extends Anasayfa {
    urunler = ["çekiç", "balta", "çivi", "masa", "sunta", "boya", "testere", "metre"];
    kampanyaliUrun;

    constructor(urunKodu, urunAdi, urunAgirligi) {
        super(...arguments);
    }


    genel_ozellikleri_goster() {
        console.log(`Bu yapı malzemesinin ürün kodu ${this.urunKodu}, ürün adı ${this.urunAdi} , ürün ağırlığı: ${this.urunAgirligi} kilogram, ürünün menşei ${this.urununMensei} olarak gözükmektedir`);
    }

    nasil_satilacak() {
        for (let i of this.urunler) {

            if (this.urunler[i] == "çekiç" || this.urunler[i] == "çivi" || this.urunler[i] == "sunta") {
                this.kampanyaliUrun = this.urunler[i];
                console.log("Yapı malzemesi bölümünün " + this.kampanyaliUrun + " kategorisinde 20.01.2023 tarhine kadar kampanya var");
            }
        }
    }
}